### PR TITLE
Tally fetch refactor

### PR DESCRIPTION
**Pull request by pettinarip**
- Decoupled the tally fetch to its own action
- Refactored the claim button to manage the tally action + loading spinner to the user

### CODE REVIEW

File: vue-app/src/components/ClaimButton.vue [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <template>
   <div>
+    <loader v-if="isLoading" />
```
Comment (pettinarip) [inline vue-app/src/components/ClaimButton.vue:3]: Main UI change

File: vue-app/src/store/actions.ts [hunk 2]
```diff
@@ -92,8 +93,11 @@ const actions = {
     //TODO: update to take factory address as a parameter, default to env. variable
     const round = await getRoundInfo(roundAddress)
     commit(SET_CURRENT_ROUND, round)
-    if (round && round.status === RoundStatus.Finalized) {
-      const tally = await getTally(roundAddress)
+  },
+  async [LOAD_TALLY]({ commit, state }) {
```
Comment (pettinarip) [inline vue-app/src/store/actions.ts:97]: We were fetching the tally file from the `LOAD_ROUND_INFO` action. Now, it has its own action which we call from the Claim button if necessary.

Comment (daodesigner) [reply vue-app/src/store/actions.ts:97]: Confirm this was tested, otherwise lgtm!

Comment (pettinarip) [reply vue-app/src/store/actions.ts:97]: Yes, it was tested.

The only place where we are using the tally file is in the Claim button component context. What I mean is that there are not a lot of places where this change can have an impact, just one.

**Review by samajammin — APPROVED** [review]

**Pull request merged**